### PR TITLE
add grpc building script.

### PR DESCRIPTION
**Pull request by Amenocy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,7 +10,7 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
-
+package-lock.json
```
Comment (XavierGeerinck) [inline .gitignore:13]: this should be commited

File: examples/http/hello-world/package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "typescript": "^4.2.4"
   },
   "dependencies": {
-    "@roadwork/dapr-js-sdk": "file:../../../build",
+    "@roadwork/dapr-js-sdk": "../../../src",
```
Comment (XavierGeerinck) [inline examples/http/hello-world/package.json:22]: Why did you change this back to src?

File: examples/http/hello-world/src/index.ts [hunk 3]
```diff
@@ -7,7 +7,9 @@ const daprAppId = "example-hello-world";
 
 async function start() {
   const client = new Dapr(daprHost, daprPort, daprInternalServerPort);
-
+  await client.pubsub.subscribe("pubsub","test",async (req,res)=>{
```
Comment (XavierGeerinck) [inline examples/http/hello-world/src/index.ts:10]: I would prefer a new example for this :) maybe in http/pubsub ?

File: grpc/.gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+node_modules/
```
Comment (XavierGeerinck) [inline grpc/.gitignore:1]: No need for this .gitignore, the main one takes care of it

File: src/http/lib/WebServer/WebServerSingleton.ts [hunk 5]
```diff
@@ -14,21 +14,21 @@ export default class WebServerSingleton {
    * This implementation let you subclass the Singleton class while keeping
    * just one instance of each subclass around.
    */
-  public static async getInstance(): Promise<WebServer> {
+  public static async getInstance(withoutInit:boolean=false): Promise<WebServer> {
```
Comment (XavierGeerinck) [inline src/http/lib/WebServer/WebServerSingleton.ts:17]: I wouldn't introduce a new parameter. Instead I would rather create an init method that has to be called to define that the Dapr Client was started. Before that we would then be able to register routes.

E.g.

```javascript
const client = new Dapr();
await client.pubsub.listen();
await client.start();
```

**Review by XavierGeerinck — CHANGES_REQUESTED** [review]

File: examples/http/hello-world/package.json (hunk 2 above)
Comment (Amenocy) [reply examples/http/hello-world/package.json:22]: i want to test changes with running Dapr instance

Comment (XavierGeerinck) [reply examples/http/hello-world/package.json:22]: I added `npm run start:dev` to the root repository, normally that should help with testing code changes in the core. For examples the `file:../../../build` should work if you run `npm run build` on the root and `npm install` on the example, which will create a symlink to the build files so they stay up-to-date whenever you rebuild the core.
